lib/dispatcher.js

```diff
@@ -74,66 +71,63 @@ class DispatcherClient {
     );
   }
 
-  #makeRequest(logContext, url, method, requestBody, cb) {
+  async #makeRequest(logContext, url, method, requestBody, cb) {
+    const requestId = uuid();
     // version *must* be provided
     const urlWithVersion = new URL(url);
     urlWithVersion.searchParams.append('version', this.#version);
     url = urlWithVersion.toString();
-    let body = '';
-    let statusCode = -1;
-    let headers = {};
-    request({
-      url,
-      method,
-      headers: {
-        'Content-Type': 'application/vnd.api+json',
-        Connection: 'Keep-Alive',
-        'Keep-Alive': 'timeout=60, max=10',
-      },
-      body: JSON.stringify(requestBody),
-    })
-      .on('error', (e) => {
+    try {
+      const response = await axiosInstance.request({
+        url,
+        method,
+        data: requestBody ?? JSON.stringify(requestBody),
```

Comment:
wouldn't that be stringifying only null or undefined here?

Reply:
`The nullish coalescing (??) operator is a logical operator that returns its right-hand side operand when its left-hand side operand is [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null) or [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined), and otherwise returns its left-hand side operand.`

Reply:
Whoops, I will fix that up